toolsrc/src/vcpkg/vcpkgpaths.cpp

```diff
@@ -456,6 +456,70 @@ If you wish to silence this error and use classic mode, you can:
         return m_pimpl->m_tool_cache->get_tool_version(*this, tool);
     }
 
+    System::ExitCodeAndOutput VcpkgPaths::run_git_command(const std::string& dot_git_dir,
+                                                          const std::string& work_tree,
+                                                          const std::string& cmd_line) const
+    {
+        const fs::path& git_exe = get_tool_exe(Tools::GIT);
+
+        System::CmdLineBuilder builder;
+        builder.path_arg(git_exe)
+            .string_arg(Strings::concat("--git-dir=", fs::u8string(dot_git_dir)))
```

Comment:
```suggestion
            .string_arg(Strings::concat("--git-dir=", dot_git_dir))
```
`dot_git_dir` and `work_tree` aren't `fs::path`s.

Reply:
Fixed, made parameters `fs::path`